lib/topic_query.rb

```diff
@@ -892,13 +892,22 @@ def remove_muted_tags(list, user, opts = nil)
           list # if viewing the topic list for a muted tag, show all the topics
         else
           muted_tag_ids = TagUser.lookup(user, :muted).pluck(:tag_id)
```

Comment:
the vast majority of the time this will return `[]` so no need to compose anything extra to the relation.

Reply:
I refactored the whole method, it should be better now. :)